parser_library/src/context/instruction_details.thh

```diff
@@ -0,0 +1,1636 @@
+DEFINE_INSTRUCTION_FORMAT(RRF_a_3, mach_format::RRF_a, reg_4_U, reg_4_U, reg_4_U)
+DEFINE_INSTRUCTION_FORMAT(RRF_a_4, mach_format::RRF_a, reg_4_U, reg_4_U, reg_4_U, mask_4_U)
+DEFINE_INSTRUCTION_FORMAT(RRF_a_4_opt, mach_format::RRF_a, reg_4_U, reg_4_U, reg_4_U_opt, mask_4_U_opt)
+DEFINE_INSTRUCTION_FORMAT(RRF_b_3, mach_format::RRF_b, reg_4_U, reg_4_U, reg_4_U)
+DEFINE_INSTRUCTION_FORMAT(RRF_b_4, mach_format::RRF_b, reg_4_U, reg_4_U, reg_4_U, mask_4_U)
+DEFINE_INSTRUCTION_FORMAT(RRF_b_4_opt, mach_format::RRF_b, reg_4_U, reg_4_U, reg_4_U, mask_4_U_opt)
+DEFINE_INSTRUCTION_FORMAT(RRF_c_3, mach_format::RRF_c, reg_4_U, reg_4_U, mask_4_U)
+DEFINE_INSTRUCTION_FORMAT(RRF_c_3_opt, mach_format::RRF_c, reg_4_U, reg_4_U, mask_4_U_opt)
+DEFINE_INSTRUCTION_FORMAT(RRF_d_3, mach_format::RRF_d, reg_4_U, reg_4_U, mask_4_U)
+DEFINE_INSTRUCTION_FORMAT(RRF_e_3, mach_format::RRF_e, reg_4_U, mask_4_U, reg_4_U)
+DEFINE_INSTRUCTION_FORMAT(RRF_e_4, mach_format::RRF_e, reg_4_U, mask_4_U, reg_4_U, mask_4_U)
+DEFINE_INSTRUCTION_FORMAT(RRS_4, mach_format::RRS, reg_4_U, reg_4_U, mask_4_U, db_12_4_U)
+DEFINE_INSTRUCTION_FORMAT(RS_a_2, mach_format::RS_a, reg_4_U, db_12_4_U)
+DEFINE_INSTRUCTION_FORMAT(RS_a_3, mach_format::RS_a, reg_4_U, reg_4_U, db_12_4_U)
+DEFINE_INSTRUCTION_FORMAT(RS_b_3, mach_format::RS_b, reg_4_U, mask_4_U, db_12_4_U)
+DEFINE_INSTRUCTION_FORMAT(RSI_3, mach_format::RSI, reg_4_U, reg_4_U, rel_addr_imm_16_S)
+DEFINE_INSTRUCTION_FORMAT(RSL_a_1, mach_format::RSL_a, db_12_4x4L_U)
+DEFINE_INSTRUCTION_FORMAT(RSL_b_3, mach_format::RSL_b, reg_4_U, db_12_8x4L_U, mask_4_U)
+DEFINE_INSTRUCTION_FORMAT(RSY_a_3, mach_format::RSY_a, reg_4_U, reg_4_U, db_20_4_S)
+DEFINE_INSTRUCTION_FORMAT(RSY_b_3_su, mach_format::RSY_b, reg_4_U, db_20_4_S, mask_4_U)
+DEFINE_INSTRUCTION_FORMAT(RSY_b_3_us, mach_format::RSY_b, reg_4_U, mask_4_U, db_20_4_S)
+DEFINE_INSTRUCTION_FORMAT(RSY_b_3_ux, mach_format::RSY_b, reg_4_U, mask_4_U, dxb_20_4x4_S)
+DEFINE_INSTRUCTION_FORMAT(RX_a_2_ux, mach_format::RX_a, reg_4_U, dxb_12_4x4_U)
+DEFINE_INSTRUCTION_FORMAT(RX_b_2, mach_format::RX_b, mask_4_U, dxb_12_4x4_U)
+DEFINE_INSTRUCTION_FORMAT(RXE_2, mach_format::RXE, reg_4_U, dxb_12_4x4_U)
+DEFINE_INSTRUCTION_FORMAT(RXE_3_xm, mach_format::RXE, reg_4_U, dxb_12_4x4_U, mask_4_U)
+DEFINE_INSTRUCTION_FORMAT(RXF_3_x, mach_format::RXF, reg_4_U, reg_4_U, dxb_12_4x4_U)
+DEFINE_INSTRUCTION_FORMAT(RXY_a_2, mach_format::RXY_a, reg_4_U, dxb_20_4x4_S)
+DEFINE_INSTRUCTION_FORMAT(RXY_b_2, mach_format::RXY_b, mask_4_U, dxb_20_4x4_S)
+DEFINE_INSTRUCTION_FORMAT(S_0, mach_format::S)
+DEFINE_INSTRUCTION_FORMAT(S_1_u, mach_format::S, db_12_4_U)
+DEFINE_INSTRUCTION_FORMAT(S_1_s, mach_format::S, db_20_4_S)
+DEFINE_INSTRUCTION_FORMAT(SI_1, mach_format::SI, db_12_4_U)
+DEFINE_INSTRUCTION_FORMAT(SI_2_s, mach_format::SI, db_12_4_U, imm_8_S)
+DEFINE_INSTRUCTION_FORMAT(SI_2_u, mach_format::SI, db_12_4_U, imm_8_U)
+//DEFINE_INSTRUCTION_FORMAT(DIAGNOSE, mach_format::DIAGNOSE)
```

Comment:
Yes, because HLASM actually does not accept it as an instruction name, it is more of a class of undocumented instructions.
They are mentioned in the PoP, so I thought I mentioned in the list as well for completeness purposes, but commented out.